### PR TITLE
Fix split worker deadlock

### LINKED ISSUE #675: Deadlock in CheckerPool when maxKeepGoingSplits is used
There is a deadlock in the CheckerPool when maxKeepGoingSplits is used and the number of cores is set to 1.
You can reproduce the issue by running the following command:
```
dotnet BoogieDriver.dll /vcsMaxKeepGoingSplits:2 /trace /proverOpt:O:smt.qi.eager_threshold=100 /vcsDumpSplits test0/MaxKeepGoingSplits.bpl
```
Boogie deadlocks on waiting to acquire `checkersSemaphore`.
When you add `/vcsCores:3` the problem disappears.

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Fixes #675

### CODE REVIEW

File: Test/.lit_test_times.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+1.166752e+00 test0/MaxKeepGoingSplits.bpl
```
Comment (shazqadeer) [inline Test/.lit_test_times.txt:1]: Did this file get checked in by mistake?

File: Test/test0/MaxKeepGoingSplits.bpl [hunk 2]
```diff
@@ -1,10 +1,12 @@
+// First line checks for a deadlock
+// RUN: %boogie /vcsMaxKeepGoingSplits:2 /proverOpt:O:smt.qi.eager_threshold=100 "%s"
 // RUN: %boogie /vcsCores:3 /vcsMaxKeepGoingSplits:2 /trace /proverOpt:O:smt.qi.eager_threshold=100 "%s" > %t.log
 // RUN: %OutputCheck --file-to-check "%t.log" "%s"
 // CHECK-L: checking split 1/1, 0.00%
```
Comment (shazqadeer) [inline Test/test0/MaxKeepGoingSplits.bpl:5]: Might it be better to make lines 5 and 6 consistent with 7-9 by dropping % and cost?

**Review by shazqadeer — APPROVED** [review]
Thanks for the timely fix.

**Comment by keyboardDrummer** [comment]
Hey, I had set this PR to automerge so it merged once you approved it, and I managed to miss your comments afterwards. I'll address them in a follow-up PR.

**Pull request merged**